tests/assembly/rust-abi-arg-attr.rs

```diff
@@ -0,0 +1,55 @@
+//@ assembly-output: emit-asm
+//@ revisions: riscv64 riscv64-zbb loongarch64
+//@ compile-flags: -C opt-level=3
+//@ [riscv64] only-riscv64
+//@ [riscv64] compile-flags: --target riscv64gc-unknown-linux-gnu
+//@ [riscv64] needs-llvm-components: riscv
+//@ [riscv64-zbb] only-riscv64
+//@ [riscv64-zbb] compile-flags: --target riscv64gc-unknown-linux-gnu
+//@ [riscv64-zbb] compile-flags: -C target-feature=+zbb
+//@ [riscv64-zbb] needs-llvm-components: riscv
+//@ [loongarch64] only-loongarch64
+//@ [loongarch64] compile-flags: --target loongarch64-unknown-linux-gnu
+//@ [loongarch64] needs-llvm-components: loongarch
```

Comment:
This should be a no_core test.

Reply:
The test file uses `{i32,u32}::max`, if marking it with `#![no_core]`, how do I use the `max` methods?

Reply:
`::gt` is also requires core.

```
error[E0369]: binary operation `>` cannot be applied to type `u32`
  --> /rust/tests/assembly/rust-abi-arg-attr.rs:39:10
   |
39 |     if a > b { a } else { b }
   |        - ^ - u32
   |        |
   |        u32

error[E0369]: binary operation `>` cannot be applied to type `i32`
  --> /rust/tests/assembly/rust-abi-arg-attr.rs:59:10
   |
59 |     if a > b { a } else { b }
   |        - ^ - i32
   |        |
   |        i32
```

Reply:
The impl of Ord for these types looks like this:

```rust
    macro_rules! ord_impl {
        ($($t:ty)*) => ($(
            #[stable(feature = "rust1", since = "1.0.0")]
            impl PartialOrd for $t {
                #[inline]
                fn partial_cmp(&self, other: &$t) -> Option<Ordering> {
                    Some(crate::intrinsics::three_way_compare(*self, *other))
                }
                #[inline(always)]
                fn lt(&self, other: &$t) -> bool { (*self) < (*other) }
                #[inline(always)]
                fn le(&self, other: &$t) -> bool { (*self) <= (*other) }
                #[inline(always)]
                fn ge(&self, other: &$t) -> bool { (*self) >= (*other) }
                #[inline(always)]
                fn gt(&self, other: &$t) -> bool { (*self) > (*other) }
            }

            #[stable(feature = "rust1", since = "1.0.0")]
            impl Ord for $t {
                #[inline]
                fn cmp(&self, other: &$t) -> Ordering {
                    crate::intrinsics::three_way_compare(*self, *other)
                }
            }
        )*)
    }
```

Reply:
Personally, assuming no objections, I am happy with this not being a no-core test until https://github.com/rust-lang/rust/pull/130693 lands.

Please leave a FIXME though.

Reply:
Oh, so if I understand correctly, if I choose to mark `no_core` in this PR, I have to copy `three_way_compare`'s declaration (including `enum Ordering`) to here, right? I'll try it after I wake up.

It looks like #130693 didn't add `Ord` for `minicore` either, should we remind them?

Reply:
Yes, exactly.

And sure, cc @jieyouxu you will want to absorb this PR, I think.

It's probably omitted because nobody's bothered so far.

Reply:
https://github.com/rust-lang/rust/pull/130693 didn't add the full core stubs because I was trying to figure out the infra first, it's intended that people can add grow the core stubs as needed (I didn't want to figure out all the core stubs). I'll absorb the ord stuff into what I had previously, but it doesn't need to block this PR (unless you want to wait for `minicore` to land first).

Reply:
Done! Added a minimal reimplementation of `{u,i}32::max`, so we can go no_core now~